### PR TITLE
【Comm】switch c_allgather in fluid to all_gather in phi

### PR DESCRIPTION
**Pull request by liym27**
### PR Category

Communication Library

### PR Types

Improvements

### Description

1. fix comm bug for new_kernel: all_gather
2. replace `c_allgather` with `all_gather` in python api

NEXT:
1. replace `c_allgather` with `all_gather` everywhere
  - https://github.com/PaddlePaddle/Paddle/pull/66338
  - https://github.com/PaddlePaddle/Paddle/pull/66538
3. delete old op `c_allgather` 

PCard-85618

### CODE REVIEW

**Review by zhangbo9674 — APPROVED** [review]
LGTM for instruction_utils

File: python/paddle/distributed/communication/stream/all_gather.py [hunk 1]
```diff
@@ -101,16 +101,17 @@ def _all_gather_in_static_mode(tensor_list, tensor, group, sync_op):
 
     ring_id = 0 if group is None else group.id
     nranks = dist.get_world_size()
-    helper.append_op(
+    op = helper.append_op(
         type=op_type,
-        inputs={'X': [tensor]},
-        outputs={'Out': [out]},
+        inputs={'x': [tensor]},
```
Comment (zhangbo9674) [inline python/paddle/distributed/communication/stream/all_gather.py:106]: op_compat 没有同步修改，这里可以改么？

Comment (liym27) [reply python/paddle/distributed/communication/stream/all_gather.py:106]: 这个算子，可以的。

**Review by heavyrain-lzy — APPROVED** [review]
LGTM for YAML

**Pull request merged**